R/load_data.R

```diff
@@ -45,29 +45,32 @@ ReadMPX_counts <- function (
   # Unzip pxl file
   original_filename <- filename
   if (endsWith(filename, ".pxl")) {
-    # LL: Returns error message
-    filename <- tryCatch(unzip(filename, "adata.h5ad", exdir = tempdir()),
+    filename <- tryCatch(utils::unzip(filename, "adata.h5ad", exdir = tempdir()),
                          error = function(e) e,
                          warning = function(w) w)
-    if (inherits(x = filename, what = "simpleWarning")) abort("Failed to unzip data")
+    if (inherits(x = filename, what = "simpleWarning"))
+      abort("Failed to unzip 'adata.h5ad' data")
   } else {
     abort(glue("Invalid file format .{.file_ext(filename)}. Expected a .pxl file."))
   }
 
   # Read temporary file
-  anndata_hier <- h5read(filename, "/")
-
-  # Remove temporary file
-  if (endsWith(original_filename, ".pxl")) file.remove(filename)
+  hd5_object <- hdf5r::H5File$new(filename, "r")
+  #anndata_hier <- h5read(filename, "/")
 
   # Extract contents
-  X <- anndata_hier$X
-  colnames(X) <- anndata_hier$obs$component
-  rownames(X) <- anndata_hier$var$marker
+  X <- hd5_object[["X"]]$read()
+  colnames(X) <- hd5_object[["obs"]][["component"]]$read()
+  rownames(X) <- hd5_object[["var"]][["marker"]]$read()
+  # X <- anndata_hier$X
+  # colnames(X) <- anndata_hier$obs$component
+  # rownames(X) <- anndata_hier$var$marker
+  X <- X[seq_len(nrow(X)), seq_len(ncol(X))]
```

Comment:
Any reason to use `seq_len(nrow(X))` rather than `1:nrow(X)`?

Reply:
It's generally [adviced](https://win-vector.com/2018/02/19/r-tip-use-seq_len-to-avoid-the-backwards-sequence-bug/#:~:text=Use%20seq_len()%20to%20avoid%20the%20backwards%20sequence%20trap.&text=Notice%20the%20above%20example%20built,x)%20%E2%80%9D%20is%20often%20wrong.) to use seq_len and seq_along instead of 1:X. In this case it shouldn't matter, but I have tried to be consistent with it :-)

Reply:
Ok, check!